network_analysis/nbconverted/2B_community_detection_louvain.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Run Louvain community detection experiments
+# 
+# To make sure that the results in `2A_community_detection_planted_partition.ipynb` aren't dependent on the choice of algorithm, this script runs the same analysis using the [Louvain community detection algorithm](https://en.wikipedia.org/wiki/Louvain_method). Here, we use the implementation in the [python-igraph package](https://igraph.org/python/doc/igraph.Graph-class.html#community_multilevel).
+# 
+# The Louvain algorithm optimizes [modularity](https://en.wikipedia.org/wiki/Louvain_method#Modularity_optimization), so in that sense it tends to find assortative (well-connected) communities similar to the PP model.
+
+# In[1]:
+
+
+import os
+import random
+
+import numpy as np
+import pandas as pd
+import igraph as ig
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches 
+import seaborn as sns
+
+from sample_nodes import (
+    sample_degree_matched,
+    sort_by_degree,
+)
+
+random.seed(1) # igraph uses Python RNG
+np.random.seed(1)
+
+
+# In[2]:
+
+
+# relevant file paths
+data_dir = './data'
+edge_list = os.path.join(data_dir, 'edge_list_processed_unsigned.csv')
+
+# map of Pa gene names to generic/not generic status, generated by Alex
+generic_gene_map = os.path.join('..', 'pseudomonas_analysis', 'annot_df.tsv')
+
+# script parameters
+NUM_NODE_SAMPLES = 1000 # number of degree-matched node samples for permutation test
+NUM_BINS = 100 # number of bins to divide nodes into, for sampling
+
+
+# In[3]:
+
+
+graph_df = pd.read_csv(edge_list)
+graph_df.head()
+
+
+# In[4]:
+
+
+G = ig.Graph.TupleList(graph_df.values,
+                       weights=True,
+                       directed=False)
+
+
+# In[5]:
+
+
+# make sure vertex/edge properties exist
+print(G.es['weight'][:5])
+
+
+# In[6]:
+
+
+annot_df = pd.read_csv(generic_gene_map, sep='\t', index_col=0)
+annot_df.head()
+
+
+# In[7]:
+
+
+is_generic = [int(annot_df.loc[v['name'], 'label']) for v in G.vs]
```

Comment:
What are you doing here?

Reply:
This just reads the generic/not generic labels from the tsv file you pointed me to into a list, in the same order as the nodes (genes) in the graph. I'll add a comment saying this.